api/src/main/java/com/foro/api/service/TopicReplyService.java

```diff
@@ -0,0 +1,125 @@
+package com.foro.api.service;
+
+import com.foro.api.reply.ReplyDTO;
+import com.foro.api.reply.ReplyRepository;
+import com.foro.api.topic.TopicDTO;
+import com.foro.api.topic.TopicRepository;
+import com.foro.api.topic.TopicStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+@Service
+public class TopicReplyService {
+    @Autowired
+    private TopicRepository topicRepository;
+    @Autowired
+    private ReplyRepository replyRepository;
+
+    @Transactional(readOnly = true)
+    public List<Topic> topicList() {
+        List<TopicDTO> topicDTOList = topicRepository.findAll();
+        List<Topic> topicList = topicDTOList.stream().map(topicDTO -> Topic.from(topicDTO)).collect(Collectors.toList());
+        return topicList;
+    }
+
+    @Transactional(readOnly = true)
+    public Topic topicById(Integer id) {
+        TopicDTO topicDTO = topicRepository.findById(id).orElse(null);
```

Comment:
Please write a test in which you try to find a topic that doesn't exist, the code should fail with a NullPointer Exception, please fix that as well.